extract.py

```diff
@@ -52,34 +69,68 @@ def get_plant_names(data: dict) -> dict:
     return {key: data.get(key) for key in relevant_cols}
 
 
+def get_origin_data(data: dict) -> dict:
+    """Returns origin data as dictionary."""
+
+    if 'origin_location' in data:
+        origin = data.get('origin_location')
+
+        lat, lon = origin[0], origin[1]
+        town, country_code = origin[2], origin[3]
+
+        region = origin[-1].split('/')
+        country, capital = region[0], region[1]
+
+        origin_dict = {
+            'lat': lat, 'lon': lon,
+            'town': town, 'country_code': country_code,
+            'country': country, 'capital': capital
+        }
+
+        return origin_dict
+
+    print('No origin location data.')
+    return None
+
+
 if __name__ == "__main__":
-    recording_data_list = []
-    watering_data_list = []
+
+    recording_data_list, watering_data_list = [], []
+
+    # Static data
     botanist_data_list = []
     plant_data_list = []
-    location_data_list = []
+    origin_data_list = []
+    image_data_list = []
+
+    with requests.Session() as session:
+        for plant_id in range(NO_OF_PLANTS):
 
-    for plant_id in range(10):
-        data = get_plant_data(plant_id)
+            print(f'Retrieving data for plant {plant_id}')
 
-        recording_data = get_recording_data(data)
-        watering_data = get_watering_data(data)
-        botanist_data = get_botanist_data(data)
-        plant_data = get_plant_names(data)
+            data = get_plant_data(plant_id, session)
 
-        recording_data_list.append(recording_data)
-        watering_data_list.append(watering_data)
-        botanist_data_list.append(botanist_data)
-        plant_data_list.append(plant_data)
+            recording_data = get_recording_data(data)
+            watering_data = get_watering_data(data)
+            botanist_data = get_botanist_data(data)
+            plant_data = get_plant_names(data)
+            origin_data = get_origin_data(data)
 
-        print(f'Retrieving data for plant {plant_id}')
+            recording_data_list.append(recording_data)
+            watering_data_list.append(watering_data)
+            botanist_data_list.append(botanist_data)
+            plant_data_list.append(plant_data)
+            origin_data_list.append(origin_data)
```

Comment:
since no processing is done inbetween, could do, say, 

recording_data_list.append(get_recording_data(data))

etc

Reply:
Yes agreed! Was also thinking of doing as list comprehensions... may refactor later on